src/jdk.internal.md/share/classes/jdk/internal/markdown/MarkdownTransformer.java

```diff
@@ -0,0 +1,951 @@
+                case EXCEPTION -> m.at(tree.pos).newExceptionTree(tree.name, desc2);
+                case THROWS -> m.at(tree.pos).newThrowsTree(tree.name, desc2);
+                default -> throw new IllegalArgumentException(tree.getKind().toString());
+            };
+        }
+
+        private DCTree.DCUnknownBlockTag transform(DCTree.DCUnknownBlockTag tree) {
+            var cont2 = transform(tree.content);
+            return (equal(cont2, tree.content))
+                    ? tree
+                    : m.at(tree.pos).newUnknownBlockTagTree(tree.name, cont2);
+        }
+
+        private DCTree.DCUnknownInlineTag transform(DCTree.DCUnknownInlineTag tree) {
+            var cont2 = transform(tree.content);
+            return (equal(cont2, tree.content))
+                    ? tree
+                    : m.at(tree.pos).newUnknownInlineTagTree(tree.name, cont2).setEndPos(tree.getEndPos());
+        }
+
+        private DCTree.DCUses transform(DCTree.DCUses tree) {
+            var desc2 = transform(tree.description);
+            return (equal(desc2, tree.description))
+                    ? tree
+                    : m.at(tree.pos).newUsesTree(tree.serviceType, desc2);
+        }
+
+        private DCTree.DCVersion transform(DCTree.DCVersion tree) {
+            var body2 = transform(tree.body);
+            return (equal(body2, tree.body))
+                    ? tree
+                    : m.at(tree.pos).newVersionTree(body2);
+        }
+
+        /**
+         * Shallow "equals" for two doc tree nodes.
+         *
+         * @param <T> the type of the items
+         * @param item1 the first item
+         * @param item2 the second item
+         * @return {@code true} if the items are reference-equal, and {@code false} otherwise
+         */
+        private static <T extends DocTree> boolean equal(T item1, T item2) {
+            return item1 == item2;
+        }
+
+        /**
+         * Shallow "equals" for two lists of doc tree nodes.
+         *
+         * @param <T> the type of the items
+         * @param list1 the first item
+         * @param list2 the second item
+         * @return {@code true} if the items are reference-equal, and {@code false} otherwise
+         */
+        private static <T extends DocTree> boolean equal(List<? extends T> list1, List<? extends T> list2) {
+            if (list1 == null || list2 == null) {
+                return (list1 == list2);
+            }
+
+            if (list1.size() != list2.size()) {
+                return false;
+            }
+
+            var iter1 = list1.iterator();
+            var iter2 = list2.iterator();
+            while (iter1.hasNext()) {
+                var item1 = iter1.next();
+                var item2 = iter2.next();
+                if (item1 != item2) {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+    }
+
+    private static final String AUTOREF_PREFIX = "code:";
```

Comment:
I guess I don't see this as being as big a deal as you seem to think it is. What is it that you are so concerned about?

I also think it is a potential feature to document and use reference links with `code:` URLs, using the reference link syntax to avoid having long method signatures in narrative text.

For example, 
```
   One of the methods on [List] has [lots of args][List.of10].
   
   [List.of10]code:List.of(E,E,E,E,E,E,E,E,E,E)
```

Reply:
I've changed the prefix to be dynamically generated. It's now called `autorefScheme` and is passed around as needed. It contains a hex hashcode, so is reasonably unguessable.

I'm still sort-of sad to lose the ability to use `code:` URLs directly, so I've left a comment in the code hinting that that is a possibility.